### PR TITLE
(LK-C-7) Add controlled adjoint support for Lightning Kokkos

### PR DESCRIPTION
**Pull request by josephleekl**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
This PR adds support for adjoint for controlled operations to Lightning Kokkos.

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

[sc-76779]

### CODE REVIEW

**Review by AmintorDusko — APPROVED** [review]
Looks good!

**Review by AmintorDusko — APPROVED** [review]
Double approved!

File: pennylane_lightning/core/src/simulators/lightning_kokkos/algorithms/AdjointJacobianKokkos.hpp [hunk 1]
```diff
@@ -150,11 +150,19 @@ class AdjointJacobian final
...
-                            mu, ops.getOpsName()[op_idx],
-                            ops.getOpsWires()[op_idx],
-                            !ops.getOpsInverses()[op_idx]) *
-                        (ops.getOpsInverses()[op_idx] ? -1 : 1);
+                        (ops.getOpsControlledWires()[op_idx].empty())
+                            ? BaseType::applyGenerator(
+                                  mu, ops.getOpsName()[op_idx],
+                                  ops.getOpsWires()[op_idx],
+                                  !ops.getOpsInverses()[op_idx]) *
+                                  (ops.getOpsInverses()[op_idx] ? -1 : 1)
```
Comment (LuisAlfredoNu) [inline pennylane_lightning/core/src/simulators/lightning_kokkos/algorithms/AdjointJacobianKokkos.hpp:158]: Here you are mixing bool operation with int. Try to use only one type

Comment (LuisAlfredoNu) [reply pennylane_lightning/core/src/simulators/lightning_kokkos/algorithms/AdjointJacobianKokkos.hpp:158]:
```suggestion
                                  ! bool *
                                  (int value diferent from 0 or 1)
```

File: pennylane_lightning/core/src/simulators/lightning_kokkos/algorithms/AdjointJacobianKokkos.hpp [hunk 2]
```diff
@@ -150,11 +150,19 @@ class AdjointJacobian final
...
+                            : BaseType::applyGenerator(
+                                  mu, ops.getOpsName()[op_idx],
+                                  ops.getOpsControlledWires()[op_idx],
+                                  ops.getOpsControlledValues()[op_idx],
+                                  ops.getOpsWires()[op_idx],
+                                  !ops.getOpsInverses()[op_idx]) *
+                                  (ops.getOpsInverses()[op_idx] ? -1 : 1);
```
Comment (LuisAlfredoNu) [inline pennylane_lightning/core/src/simulators/lightning_kokkos/algorithms/AdjointJacobianKokkos.hpp:165]: same here

File: tests/lightning_qubit/test_adjoint_jacobian_class.py [hunk 3]
```diff
@@ -165,10 +165,6 @@ def test_not_expectation_return_type(self, lightning_sv):
         ):
             self.calculate_jacobian(lightning_sv(num_wires=1), tape)
 
-    @pytest.mark.skipif(
-        device_name in ("lightning.kokkos"),
-        reason="N-controlled operations are not implemented in lightning.kokkos.",
-    )
```
Comment (LuisAlfredoNu) [inline tests/lightning_qubit/test_adjoint_jacobian_class.py:171]: Nice!

**Review by LuisAlfredoNu — COMMENTED** [review]
Thank you @josephleekl 
A few coments

File: pennylane_lightning/core/src/simulators/lightning_kokkos/algorithms/AdjointJacobianKokkos.hpp (hunk 1 above)
Comment (josephleekl) [reply pennylane_lightning/core/src/simulators/lightning_kokkos/algorithms/AdjointJacobianKokkos.hpp:158]: Actually it's multiplying `PrecisionT` with `int`

The `!ops.getOpsInverses()[op_idx]` in L157 is an argument to `applyGenerator`, which returns `PrecisionT`, which is then multiplied by `(ops.getOpsInverses()[op_idx] ? -1 : 1)`

Comment (LuisAlfredoNu) [reply pennylane_lightning/core/src/simulators/lightning_kokkos/algorithms/AdjointJacobianKokkos.hpp:158]: ooh sorry, I missed the close bracket before `*`

**Review by LuisAlfredoNu — APPROVED** [review]
LGTM!
Thank you @josephleekl

**Comment by LuisAlfredoNu** [comment]
Please wait until all tests are green! if not tag me again. :rocket:

**Pull request merged**